src/table/use-sticky-columns.ts

```diff
@@ -0,0 +1,362 @@
+        scrollPadding: { left: this.stickyWidthLeft, right: this.stickyWidthRight },
+      }));
+    }
+  }
+
+  private updateScroll(props: UpdateCellStylesProps) {
+    const wrapperScrollLeft = props.wrapper.scrollLeft;
+    const wrapperScrollWidth = props.wrapper.scrollWidth;
+    const wrapperClientWidth = props.wrapper.clientWidth;
+    const tablePaddingLeft = parseInt(getComputedStyle(props.table).paddingLeft) || 0;
+    const tablePaddingRight = parseInt(getComputedStyle(props.table).paddingRight) || 0;
+
+    this.isStuckToTheLeft = wrapperScrollLeft > tablePaddingLeft;
+    this.isStuckToTheRight = Math.ceil(wrapperScrollLeft) < wrapperScrollWidth - wrapperClientWidth - tablePaddingRight;
+  }
+
+  private generateCellStyles = (props: UpdateCellStylesProps): Record<ColumnId, null | StickyColumnsCellState> => {
+    const isEnabled = this.isEnabled(props);
+
+    return props.visibleColumns.reduce((acc, columnId, index) => {
+      let stickySide = 'non-sticky';
+      if (index < props.stickyColumnsFirst) {
+        stickySide = 'left';
+      } else if (index >= props.visibleColumns.length - props.stickyColumnsLast) {
+        stickySide = 'right';
+      }
+
+      if (!isEnabled || stickySide === 'non-sticky') {
+        acc[columnId] = null;
+        return acc;
+      }
+
+      // Determine the offset of the sticky column using the `cellOffsets` state object
+      const isFirstColumn = index === 0;
+      const stickyColumnOffsetLeft = this.cellOffsets.get(columnId)?.first ?? 0;
+      const stickyColumnOffsetRight = this.cellOffsets.get(columnId)?.last ?? 0;
+      const cellStyle = {
+        left: stickyColumnOffsetLeft !== undefined && stickySide === 'left' ? stickyColumnOffsetLeft : undefined,
+        right: stickyColumnOffsetRight !== undefined && stickySide === 'right' ? stickyColumnOffsetRight : undefined,
+      };
+      const lastLeftStickyColumnIndex = props.stickyColumnsFirst - 1;
+      const lastRightStickyColumnIndex = props.visibleColumns.length - props.stickyColumnsLast;
+      const tablePaddingLeft = parseInt(getComputedStyle(props.table).paddingLeft) || 0;
+
+      acc[columnId] = {
+        padLeft: isFirstColumn && tablePaddingLeft !== 0 && this.isStuckToTheLeft,
+        lastLeft: this.isStuckToTheLeft && lastLeftStickyColumnIndex === index,
+        lastRight: this.isStuckToTheRight && lastRightStickyColumnIndex === index,
+        offset: cellStyle,
+      };
+      return acc;
+    }, {} as Record<ColumnId, null | StickyColumnsCellState>);
+  };
+
+  private updateCellOffsets = (props: UpdateCellStylesProps): void => {
+    const firstColumnWidths: number[] = [];
+    for (let i = 0; i < props.visibleColumns.length; i++) {
+      const element = props.cells[props.visibleColumns[i]];
+      const cellWidth = element.getBoundingClientRect().width ?? 0;
+      firstColumnWidths[i] = (firstColumnWidths[i - 1] ?? 0) + cellWidth;
+    }
+
+    const lastColumnsWidths: number[] = [];
```

Comment:
minor: one constant uses "Column*s*" plural while the other uses "Column" singular, that seems unintentional